influxql/statement_rewriter.go

```diff
@@ -149,3 +151,88 @@ func rewriteShowTagKeysStatement(stmt *ShowTagKeysStatement) (Statement, error)
 		Dedupe:     true,
 	}, nil
 }
+
+func rewriteShowTagValuesStatement(stmt *ShowTagValuesStatement) (Statement, error) {
+	// Check for time in WHERE clause (not supported).
+	if HasTimeExpr(stmt.Condition) {
+		return nil, errors.New("SHOW TAG VALUES doesn't support time in WHERE clause")
+	}
+
+	condition := stmt.Condition
+	if len(stmt.Sources) > 0 {
+		if source, ok := stmt.Sources[0].(*Measurement); ok {
+			var expr Expr
+			if source.Regex != nil {
+				expr = &BinaryExpr{
+					Op:  EQREGEX,
+					LHS: &VarRef{Val: "name"},
+					RHS: &RegexLiteral{Val: source.Regex.Val},
+				}
+			} else if source.Name != "" {
+				expr = &BinaryExpr{
+					Op:  EQ,
+					LHS: &VarRef{Val: "name"},
+					RHS: &StringLiteral{Val: source.Name},
+				}
+			}
+
+			// Set condition or "AND" together.
+			if condition == nil {
+				condition = expr
+			} else {
+				condition = &BinaryExpr{Op: AND, LHS: expr, RHS: condition}
+			}
+		}
+	}
+
+	if len(stmt.TagKeys) > 0 {
+		var expr Expr
+		for _, tagKey := range stmt.TagKeys {
+			if expr != nil {
+				expr = &BinaryExpr{
+					Op:  OR,
+					LHS: expr,
+					RHS: &BinaryExpr{
+						Op:  EQ,
+						LHS: &VarRef{Val: "_tagKey"},
+						RHS: &StringLiteral{Val: tagKey},
+					},
+				}
+			} else {
+				expr = &BinaryExpr{
+					Op:  EQ,
+					LHS: &VarRef{Val: "_tagKey"},
+					RHS: &StringLiteral{Val: tagKey},
+				}
```

Comment:
Can you create the `_tagKey = value` expression above the `if` so you can reuse it in the `if` & `else` blocks?